packages/i18n/src/i18n.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * External dependencies
+ */
+import Tannin from 'tannin';
+
+/**
+ * @typedef {{[key: string]: any}} LocaleData
+ */
+
+/**
+ * Default locale data to use for Tannin domain when not otherwise provided.
+ * Assumes an English plural forms expression.
+ *
+ * @type {LocaleData}
+ */
+const DEFAULT_LOCALE_DATA = {
+	'': {
+		plural_forms: ( n ) => ( n === 1 ? 0 : 1 ),
+	},
+};
+
+/**
+ * Instantiable I18n methods
+ */
+export class I18n {
```

Comment:
I don't mind the class but it seems that the pattern used in other places in Gutenberg is "create" functions instead. Should we make it consistent? (createRegistry for instance)

Reply:
That's a good observation: the Gutenberg codebase rarely uses `new Something` to instantiate a class. There are factory functions instead.

On the other hand, the `createI18n` factory function is redundant in the sense that it doesn't add anything new: it just forwards the arguments verbatim:
```js
const createI18n = ( ...args ) => new I18n( ...args );
```
And we'll likely need to export the `I18n` class itself, too, one day, because its type is part of the public API.

Reply:
In my own prior thoughts about an implementation, I too had expected this to be implemented as `createI18n`. The note about documenting the type is interesting. We've had reasonable success with documenting object types (e.g. [`createQueue`](https://github.com/WordPress/gutenberg/blob/1b724f316151dd9219905514b0f3401e891eb9ab/packages/priority-queue/src/index.js#L59)), but a class does seem like the more "natively understood" implementation.

I guess it doesn't bother me too much to export this as a class, aside from the inconsistency.

Reply:
Coming back here, I'm wondering if we should just stay consistent, it doesn't seem like we'll be losing anything by doing so.

Reply:
Have converted the class to a `create*` function.